skimage/future/trainable_segmentation.py

```diff
@@ -0,0 +1,164 @@
+from skimage.feature import multiscale_basic_features
+
+try:
+    from sklearn.exceptions import NotFittedError
+    from sklearn.ensemble import RandomForestClassifier
+    has_sklearn = True
+except ImportError:
+    has_sklearn = False
+
+    class NotFittedError(Exception):
+        pass
+
+
+class TrainableSegmenter(object):
+    """
+    Estimator for classifying pixels.
+
+    Parameters
+    ----------
+    clf : classifier object, optional
+        classifier object, exposing a ``fit`` and a ``predict`` method as in
+        scikit-learn's API, for example an instance of
+        ``RandomForestClassifier`` or ``LogisticRegression`` classifier.
+    features_func : function, optional
+        function computing features on all pixels of the image, to be passed
+        to the classifier. The output should be of shape
+        ``(m_features, *labels.shape)``. If None,
+        :func:`skimage.segmentation.multiscale_basic_features` is used.
+
+    Methods
+    -------
+    compute_features
+    fit
+    predict
+    """
+
+    def __init__(self, clf=None, features_func=None):
+        if clf is None:
+            if has_sklearn:
+                self.clf = RandomForestClassifier(n_estimators=100, n_jobs=-1)
+            else:
+                raise ImportError(
+                    "Please install scikit-learn or pass a classifier instance"
+                    "to TrainableSegmenter."
+                )
+        else:
+            self.clf = clf
+        self.features_func = features_func
+
+    def compute_features(self, image):
+        if self.features_func is None:
+            self.features_func = multiscale_basic_features
+        self.features = self.features_func(image)
+
+    def fit(self, labels, image=None):
```

Comment:
Why `image` is set to `None` by default?

Reply:
Yes, it's puzzling... 
```suggestion
    def fit(self, labels, image):
```

Reply:
I don't remember, this is puzzling indeed. And to mimic scikit-learn's syntax, this should be `fit(self, image, labels)` (it's `fit(X_train, y_train)`). Let me change this ASAP.